tests/test_keep-alive.py

```diff
@@ -0,0 +1,69 @@
+import gc
+
+
+def test_keep_alive_argument(capture):
+    from pybind11_tests import Parent, Child
+
+    with capture:
+        p = Parent()
+    assert capture == "Allocating parent."
+    with capture:
+        p.addChild(Child())
+        gc.collect()
+    assert capture == """
+        Allocating child.
+        Releasing child.
+    """
+    with capture:
+        del p
+        gc.collect()
+    assert capture == "Releasing parent."
+
+    with capture:
+        p = Parent()
+    assert capture == "Allocating parent."
+    with capture:
```

Comment:
Seeing as this is a very common case, would it make sense to have an `assert_capture` helper? So you can do this:

```python
with assert_capture('Allocating parent.'):
    p = Parent()
```

Could shave off half a hundred lines from the test suite.